app/src/main/java/com/github/palFinderTeam/palfinder/ProfileViewModel.kt

```diff
@@ -116,10 +116,20 @@ class ProfileViewModel @Inject constructor(
      */
     fun block(userId: String, otherId: String) {
         viewModelScope.launch {
-            profileService.fetch(userId)?.let {
+            profileService.fetch(userId)?.let { user ->
                 // Sanity check
-                if (userId != otherId && !it.blockedUsers.contains(otherId)) {
-                    profileService.edit(it.uuid, BLOCKED_USERS, it.blockedUsers.plus(otherId))
+                if (userId != otherId && !user.blockedUsers.contains(otherId)) {
+                    val newBlockList = user.blockedUsers.plus(otherId)
+                    profileService.unfollowUser(user, otherId)
+                    profileService.edit(user.uuid, BLOCKED_USERS, newBlockList)
+                    // Leave all meetups organize by blocked user
+                    for (meetupId in user.joinedMeetUps) {
```

Comment:
Isn't that a bit extreme? Might need to show the user a confirmation before leaving all meetup

Reply:
I think it's okay because user can't join meetup created by someone they blocked, maybe just add a warning to say you left x meetups

Reply:
Indeed, not an issue, won't fix